tests/end_to_end_test/end_to_end_test.go

```diff
@@ -916,6 +916,15 @@ func (e *testenv) run(t *testing.T, args ...string) ([]string, error) {
 
 	o, err := c.Output()
 
+	code := c.ProcessState.ExitCode()
+	if code != 0 && err == nil {
```

Comment:
This is testing the `os.Cmd` implementation, but not `kopia` itself.
According to the `Cmd.Output()` documentation:

> Any returned error will usually be of type *ExitError.

The way I interpret this (and I could be wrong because I'm making implicit assumptions and extrapolating from that comment) is that an error will be returned when either:
* The command fails to execute (for example, binary not found, corrupt or unreadable);
* The command "fails" by returning a non-zero exit code; or
* There is some other error while reading the commands output. This seems unlikely, as this would be included in the case where the command itself fails.

Reply:
yeah I think you're right about that; the scope of K10-3199 seems a little off then. By virtue of the `runAndExpectFailure` for `kopia [snap] restore` commands in e2e testing, we already verify the exit code. Do we narrow the scope to unit testing, and just ensure the cases we're interested in return error? Because any/all errors returned by the command execution will verify exit code